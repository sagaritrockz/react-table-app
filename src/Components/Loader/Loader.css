.loader-container {
    height: 100%;
    width: 100%;
    background-color: black;
    position: absolute;
    top: 0px;
    left: 0px;
    opacity: 0.4;
}

.loader {
    border: 5px solid #f3f3f3;
    -webkit-animation: spin 1s linear infinite;
    animation: spin 1s linear infinite;
    border-top: 5px solid #555;
    border-radius: 50%;
    width: 40px;
    height: 40px;
    position: absolute;
    top: 50%;
    left:  50%;
}
.show-loader {
    display: block !important;
}
.hide-loader {
    display: none !important; 
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
}