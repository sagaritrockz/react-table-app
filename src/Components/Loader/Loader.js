import React from "react";
import { bool } from "prop-types";
import "./Loader.css";

const Loader = ({ loading }) => {
    return (
        <div
            className={
                "loader-container " + (loading ? "show-loader" : "hide-loader")
            }
        >
            <div className="loader" />
        </div>
    );
};

Loader.defaultProps = {
    loading: false
};

Loader.propTypes = {
    loading: bool
};

export default Loader;
