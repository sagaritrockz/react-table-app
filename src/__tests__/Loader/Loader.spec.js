import React from 'react'
import Enzyme from 'enzyme'
import Adaptar from 'enzyme-adapter-react-16.3'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Loader from '../../Components/Loader/Loader'

Enzyme.configure({adapter: new Adaptar()});

describe('Loader', () => {
    it('should render correctly on showLoader', () => {
        const showLoader = true
        const output = shallow(
          <Loader loading={showLoader} />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('should render correctly on hideLoader', () => {
        const showLoader = false
        const output = shallow(
          <Loader loading={showLoader} />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});