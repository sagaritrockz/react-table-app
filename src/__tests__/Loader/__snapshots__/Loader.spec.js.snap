// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly on hideLoader 1`] = `
<div
  className="loader-container hide-loader"
>
  <div
    className="loader"
  />
</div>
`;

exports[`Loader should render correctly on showLoader 1`] = `
<div
  className="loader-container show-loader"
>
  <div
    className="loader"
  />
</div>
`;
